### ISSUE TITLE
Unexpected behavior 2 in ReactiveProperty

### ISSUE THREAD

**Issue by kimk-s** [Cysharp/R3#167]
Thanks for your library.

This only works in 1.1.8
It doesn't seem to work in 1.1.7 and 1.1.9

```csharp
var p1 = new ReactiveProperty<int>();
p1.Skip(1).Subscribe(x => Debug.Log("[P1]" + x));

var d1 = p1.Skip(1).Subscribe(x => Debug.Log("[P2]" + x));
var d2 = p1.Skip(1).Subscribe(x => Debug.Log("[P2]" + x));
d1.Dispose();
d2.Dispose();

p1.Skip(1).Subscribe(x => Debug.Log("[P3]" + x)); // P3 is not raised

p1.Value = 1;
p1.Value = 2;

public static class Debug
{
    public static void Log(string x)
    {
        Console.WriteLine(x);
    }
}
```

**Comment by neuecc** [comment]
thanks! I'll investigate soon.

**Comment by neuecc** [comment]
I've released v1.1.10.
I apologize for the multiple messages, and thank you very much!
The ad-hoc fixes were not good.
I have replaced it with a completely new implementation that includes a fundamental review and improvements to thread safety. I have also added more tests.
Your example has been included in the unit test cases.
